src/Foundation/NSFileManager.cs

```diff
@@ -425,6 +426,59 @@ internal static NSFileSystemAttributes FromDictionary (NSDictionary dict)
 	}		
 	
 	public partial class NSFileManager {
+
+		[DllImport (Constants.FoundationLibrary)]
+		static extern IntPtr NSUserName ();
+
+		public static string UserName {
+			get {
+				using (var nsstring = ObjCRuntime.Runtime.GetNSObject<NSString> (NSUserName ()))
```

Comment:
How about the new style using format for these?

Reply:
Will work the same way. using will be until the end of the get. Problem with me is that I don't trust that new using syntactic sugar. I don't trust myself with the scope ;)

Reply:
Maybe use [NSString.FromHandle](https://github.com/xamarin/xamarin-macios/blob/fffa889f22fad3c220e1b50a5884e077b0556e99/src/Foundation/NSString.cs#L196
):
```csharp
				return NSString.FromHandle (NSUserName ());
```